Sources/Target/Windows/Process.cpp

```diff
@@ -331,6 +331,18 @@ ErrorCode Process::readMemory(Address const &address, void *data, size_t length,
 
 ErrorCode Process::writeMemory(Address const &address, void const *data,
                                size_t length, size_t *nwritten) {
+  // We try to commit the necessary pages in case they have been uncommitted.
+  // Otherwise writes may fail.
+  LPVOID allocError = VirtualAllocEx(
+      _handle,
+      reinterpret_cast<LPVOID>(address.value()),
+      length,
+      MEM_COMMIT,
+      PAGE_EXECUTE_READWRITE);
+  if (allocError == NULL) {
+    return Platform::TranslateError();
+  }
+
```

Comment:
In fact, i'm changing the protection of memory, I will reuse the same protection it had before.
According to msdn, commiting an unallocated page doesn't allocate memory, the operation fails instead
I've seen that lldb sometimes tries to write some memory in a page and it fails unless the page is commited, I'll investigate more this issue